docker_mcp/core/safety.py

```
@@ -0,0 +1,244 @@
+        if not any(snap_name.startswith(prefix) for prefix in migration_prefixes):
+            return False, f"Snapshot '{snap_name}' does not appear to be migration-related"
+        
+        # Check for suspicious patterns
+        if len(snap_name) < 10:  # Migration snapshots should have timestamps
+            return False, f"Snapshot name '{snap_name}' is too short - may not be migration-related"
+        
+        return True, f"ZFS snapshot deletion validated: {snapshot_name}"
+    
+    async def safe_cleanup_archive(self, ssh_cmd: list[str], archive_path: str, reason: str = "Migration cleanup") -> tuple[bool, str]:
+        """Safely cleanup archive files with validation.
+        
```

Ruff (0.12.2)

Blank line contains whitespace

Remove whitespace from blank line

(W293)